internal/api/ring.go

```diff
@@ -281,6 +280,112 @@ func handleUpdateRing(c *Context, w http.ResponseWriter, r *http.Request) {
 	outputJSON(c, w, ring)
 }
 
+// handleReleaseAllRings responds to POST /api/rings/release,
+// releasing a deployment in all rings.
+func handleReleaseAllRings(c *Context, w http.ResponseWriter, r *http.Request) {
+	releaseRingRequest, err := model.NewReleaseRingRequestFromReader(r.Body)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to deserialize ring release request body")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	rings, err := c.Store.GetRings(&model.RingFilter{
+		IncludeDeleted: false,
+		PerPage:        10000,
+	})
+
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to get all rings from store")
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+
+	for _, ring := range rings {
```

Comment:
After discussing this with @Szymongib I moved forward with the following: 
- Lock all rings in batch with single DB query (if non of them is locked, otherwise we fail immediatly)
- Verify that all rings can be transitioned to new stat -- if at least one is not we fail and unlock all.
- Change state of all Rings in single DB query